homeworks/01-bubble-sort/src/components/SortField.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { randomNum, randomArray, delay } from './Helpers';
+import Button from './Button';
+import Column from './Column';
+
+interface SortFieldProps { };
+
+interface SortFieldState {
+    drawArr: number[],
+    isSorted: string,
+};
+
+type StepParams = [number[], boolean];
+
+const MAX_ARR_LENGTH = 15;
+const MIN_ARR_LENGTH = 10;
+const MAX_ARR_VALUE = 500;
+export class SortField extends React.Component<SortFieldProps, SortFieldState> {
+
+    constructor(props: SortFieldProps) {
+        super(props);
+        this.state = {
+            drawArr: randomArray(randomNum(MAX_ARR_LENGTH, MIN_ARR_LENGTH), MAX_ARR_VALUE),
+            isSorted: 'Not started'
+        }
+        this.onGenerateArr = this.onGenerateArr.bind(this);
+        this.onStartSorting = this.onStartSorting.bind(this);
+    }
+
+    onGenerateArr() {
+        this.setState({ drawArr: randomArray(randomNum(MAX_ARR_LENGTH, MIN_ARR_LENGTH), MAX_ARR_VALUE) });
+        this.setState({ isSorted: 'Not started' });
+    }
+
+    onStartSorting() {
+        const arr: number[] = [...this.state.drawArr];
+        this.bubbleSort(arr);
+        this.setState({ isSorted: 'Sorting' });
+    }
+
+    bubbleStep(arr: number[], j: number, isSwapped: boolean): StepParams {

```

Comment:
Better to keep a business logic separate from the component. E.g., create a pure function and move it to another file (e.g., under `utils` folder).